official/transformer/utils/dataset.py

```diff
@@ -247,13 +248,26 @@ def _read_and_batch_from_files(
   dataset = dataset.repeat(repeat)
 
   # Prefetch the next element to improve speed of input pipeline.
-  dataset = dataset.prefetch(1)
+  dataset = dataset.prefetch(buffer_size=tf.contrib.data.AUTOTUNE)
+  return dataset
+
+
+def _generate_synthetic_data(params):
+  """Create synthetic data based on the parameter batch size."""
+  batch = length = int(math.sqrt(params["batch_size"]))
+  dataset = tf.data.Dataset.from_tensors(tf.ones([batch, length], tf.int64))
+  dataset = dataset.map(lambda x: (x, x))
+  dataset = dataset.cache()
+  dataset = dataset.repeat(1000)
```

Comment:
I would prefer that we don't hard code a dummy size into the synthetic data. Because there is already support for setting a step based schedule, infinite repeat should be fine. (This is what ResNet does.)

Reply:
See similar message in #4476 , but we should make a helper fn for getting synthetic data in a particular shape... don't both do that, though.